src/System.Runtime/tests/System/DoubleTests.cs

```diff
@@ -205,6 +205,7 @@ public void CompareTo_ObjectNotDouble_ThrowsArgumentException(object value)
         [InlineData((double)789, (double)-789, false)]
         [InlineData((double)789, (double)0, false)]
         [InlineData(double.NaN, double.NaN, true)]
+        [InlineData(double.NaN, -double.NaN, true)]
```

Comment:
You're going to prevent this case from running on NetFX because on these will not hash the same on NetFX.

Reply:
@JonHanna The result will be the same `0 - NaN` still is `NaN`

Reply:
@JonHanna So should I skip this test for NetFX because we only fix this bug for .NET Core?

Reply:
@mazong1123 yes. (Do we have a way to skip an inline data attribute for a platform? I know only of how to do that for a whole test myself).

@leppie Linqpad scratchcode: `(double.NaN).GetHashCode()` gives `-524288`, `(-double.NaN).GetHashCode()` gives `2146959360`. Or indeed, the details of the coreclr bug this PR will confirm is fixed.

Reply:
@JonHanna Didnt see the hashcode call. Line 227 should be moved to the non-NaN case then.

Reply:
@leppie not after dotnet/coreclr#6237 is fixed, which this test will confirm.

Reply:
We do not have a way to skip individual inline data. You have to make a second block of inline data, that's for everything but netfx. Then have two tests, one for each, and on one of them you have 
`[SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, ".. reason..")]`

Reply:
I thought so. At least the second can just call into the first.

Reply:
@danmosemsft I'm wondering why we mix the tests for netfx and corefx together when they may have different codebase. This introduced the complexity of writing tests.

Reply:
One thing is that most things should be the same, so something working on one platform and not another is a canary in the mine. Cases like this were an issue on both is fixed in corefx but not (yet, anyway) in netfx is an obvious exception, though.

Reply:
@mazong1123 good question. It is not a goal to develop a test bed for netfx changes. When we modify netfx (eg for servicing) we do not run these tests: it has its own closed source tests. The reason we run these tests on netfx, as @JonHanna says, is that it helps us spot places where .NET Core behavior diverges from netfx behavior. So we have some tests that run on both, and some that only run on corefx. We have few or no tests that run only on netfx, as that is not relevant for our purpose. By default, all tests will run on netfx as well as corefx.

Reply:
@danmosemsft @JonHanna  Thanks for your clarification. I'll update the tests according to the feedback.